### PR TITLE
openai.LLM: add common API-compatible adapters

### PR DESCRIPTION
**Pull request by nbsp**
supports Ollama, Fireworks, Perplexity, Groq, Together, Octo.

Fireworks and Ollama accept arbitrary strings for models, checking against a list that gets updated too frequently for us to hardcode.

Ollama doesn't read the API key, but `openai` requires it regardless, so the guide says to supply it with an arbitrary blob.

Perplexity doesn't seem to need `/v1/` appended to its API URL.

currently untested, but all of these projects' documentations imply they work as a drop-in replacement.

~~**NOTE**: this is a minor bump! maybe let's not merge this until we want to release a v0.9.0~~

### CODE REVIEW

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:105]: It may not be the best name we picked for Azure, but should we keep the names consistent? cc @livekit/agent-devs

Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:105]: It may not be the best name we picked for Azure, but should we keep the names consistent? cc @livekit/agent-devs

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/models.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/models.py:57]:
```suggestion
```

**Comment by theomonnom** [comment]
Let's add https://www.together.ai/ I know Shayne is using it :) 
https://octo.ai/ may also be relevant

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py
Comment (nbsp) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:105]: i think an important difference here is that Azure is still an OpenAI usecase, while the rest are other services supporting the same API. so `create_perplexity_client` makes less sense than it does for azure imo. i'm fine with changing it if there's consensus though

**Comment by nbsp** [comment]
> Let's add [together.ai](https://www.together.ai/) I know Shayne is using it :) [octo.ai](https://octo.ai/) may also be relevant

added both of those as well!

noting something here that i just edited onto the top message: we're going to have to release 0.9.0 for this, so let's not merge this until 0.8 stabilizes

**Review by davidzhao — COMMENTED** [review]
awesome! just to resolve the consistency question

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py
Comment (nbsp) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:131]: got it. just another potential issue: do we go with `ai` at the end of the names? or not? or depending on what the provider calls itself? for example `groq`/`fireworks` vs `octo`/`together`

**Comment by theomonnom** [comment]
Why does this PR can't be a patch instead of a minor version?

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py
Comment (davidzhao) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:131]: I think without is better. AI is kind of redundant. let's also alias the `create_azure_client` to `with_azure`, while marking the former deprecated.

**Comment by nbsp** [comment]
> > noting something here that i just edited onto the top message: we're going to have to release 0.9.0 for this, so let's not merge this until 0.8 stabilizes
> 
> Why does this PR can't be a patch instead of a minor version? I know we're releasing new features but since we're not breaking anything etc.. no need to wait

your call. thinking about it from a semver compliance POV, but it doesn't matter

**Comment by nbsp** [comment]
> I think without is better. AI is kind of redundant. let's also alias the `create_azure_client` to `with_azure`, while marking the former deprecated.

https://github.com/livekit/agents/pull/611#discussion_r1712680890
if we change the function name for azure we have to bump minor anyway

**Review by davidzhao — APPROVED** [review]
lg!

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:183]: Can you use the logger from the plugin

**Comment by someshfengde** [comment]
can I use Groq with livfekit?

**Comment by billyg88** [comment]
> can I use Groq with livekit?

Try this ```with_groq``` there is a specific method for that... look at the code in ```https://github.com/livekit/agents/blob/main/livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py```

**Pull request merged**
